tests/models/llama/test_tokenization_llama.py

```diff
@@ -822,3 +825,28 @@ def test_special_tokens_strip(self):
         self.assertEqual(input_ids, [284, 1, 156])
         tokens = self.tokenizer.tokenize("No <s> ▁He")
         self.assertEqual(tokens, ["▁No", "<s>", "▁He"])  # spaces are eaten by rstrip / lstrip
+
+
+@require_read_token
+class TikTokenIntegrationTests(unittest.TestCase):
+    """
+    A class that regroups important test to make sure that we properly handle the special tokens.
+    """
+
+    def test_tiktoken_llama(self):
+        model_path = "hf-internal-testing/Llama3-Instruct-Internal"
+        model_file_name = "tokenizer.model"
+
+        tiktoken_tokenizer = AutoTokenizer.from_pretrained(model_path, tiktoken_file=model_file_name, from_slow=True)
```

Comment:
the filename is always the same, we should not need to specify it IMO!

Reply:
agreed! Would a true/false flag work? To let users indicate that they want to convert from tiktoken

Reply:
No I would rather we first try with sentencepiece, then if we have excpet TikTokenError (i don't know the name) then try tiktoken